javascript/javascript2/week1/main.js

```diff
@@ -0,0 +1,18 @@
+//find the shortest word
+const danishWords = ["bil", "plante", "kaffe", "bog", "ø", "planetarium"];
+console.log(getShortestWord(danishWords));
+function getShortestWord(danishWords) {
+  if (
+    arguments.length < 1 ||
+    !Array.isArray(danishWords) ||
+    danishWords.length === 0
+  ) {
+    return "Please provide an array with strings";
+  }
+  // not checking if all elements are string
+  let shortestWord = danishWords[0];
+  danishWords.reduce((previous, current) => {
+    if (current.length < shortestWord.length) shortestWord = current;
+  });
+  return shortestWord;
+}
```

Comment:
💫 Great Work 💫

Out of curiosity, why are you applying the reduce() method in this case? 😊 Typically, the reduce method is applied to return an accumulated result based on the values of an array. In this case, you are applying the method to call a statement on each array element, which you could achieve with a simple for loop. I would like to hear your thoughts and reasoning behind your implementation 😊💡

Great work overall 🔥

> https://www.w3schools.com/jsref/jsref_reduce.asp

Reply:
I understand reduce as a general purpose method, which can be used in many situations. (not just to calculate sum). basically it iterates through the array (equivalent to for loop) and we can do our processing as required. even filter , map etc can be written using reduce , which will be more efficient(according to developer.mozilla). so I just tried reduce to find the shortest word and it was working.

Reply:
It can most certainly be used as a general method to which you can apply a statement to each element of an array, you are correct. But in your implementation, your keyword previous, is declared but never read. Which is why I was wondering why you chose this implementation as supposed to something else. 

It is working - all is good. I merely wanted a dialogue and reflection upon this  - seems like you have a great understanding in general 😊

Reply:
yes that's true , previous is not being used .. may be I can rewrite using sort or simple for loop .. thank you